AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/DataStoreSyncExpression.swift

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+public typealias QueryPredicateResolver = () -> QueryPredicate
+
+public struct DataStoreSyncExpression {
+    let modelType: Model.Type
+    let modelPredicate: QueryPredicateResolver
+
+    static public func syncExpression(_ modelType: Model.Type,
```

Comment:
I know this is still in draft. This might break the flutter implementation where we are not relying on `Model.Type`. Can we change this to :
```swift 
static public func syncExpression(_ modelSchema: ModelSchema,
```
or
```swift
static public func syncExpression(_ modelName: ModelName,
```